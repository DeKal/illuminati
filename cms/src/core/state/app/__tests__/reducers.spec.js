import {
  APP_TURN_ON_LOADING,
  APP_TURN_OFF_LOADING,
  OPEN_TOAST,
  CLOSE_TOAST,
  ERROR
} from 'core/state/actionType'

import appReducers from '../reducers'

describe('App Reducer', () => {
  describe('default', () => {
    it('should work corectly', function() {
      const action = {
        type: 'default'
      }
      expect(appReducers({}, action)).toMatchSnapshot()
    })
  })

  describe('APP_TURN_ON_LOADING', () => {
    it('should work corectly', function() {
      const action = {
        type: APP_TURN_ON_LOADING
      }
      expect(appReducers({}, action)).toMatchSnapshot()
    })
  })

  describe('APP_TURN_OFF_LOADING', () => {
    it('should work corectly', function() {
      const action = {
        type: APP_TURN_OFF_LOADING
      }
      expect(appReducers({}, action)).toMatchSnapshot()
    })
  })

  describe('OPEN_TOAST', () => {
    it('should work corectly', function() {
      const action = {
        type: OPEN_TOAST,
        payload: {
          success: true,
          msg: 'successfully save user'
        }
      }
      expect(appReducers({}, action)).toMatchSnapshot()
    })
  })

  describe('CLOSE_TOAST', () => {
    it('should work corectly', function() {
      const action = {
        type: CLOSE_TOAST
      }
      expect(appReducers({}, action)).toMatchSnapshot()
    })
  })

  describe('ERROR', () => {
    it('should work corectly', function() {
      const action = {
        type: ERROR,
        payload: 'Some thing wrong happened'
      }
      expect(appReducers({}, action)).toMatchSnapshot()
    })
  })
})
