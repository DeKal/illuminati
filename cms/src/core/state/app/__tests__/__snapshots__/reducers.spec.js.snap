// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Reducer APP_TURN_OFF_LOADING should work corectly 1`] = `
Object {
  "loading": false,
}
`;

exports[`App Reducer APP_TURN_ON_LOADING should work corectly 1`] = `
Object {
  "loading": true,
}
`;

exports[`App Reducer CLOSE_TOAST should work corectly 1`] = `
Object {
  "toast": Object {
    "isDisplayToast": false,
  },
}
`;

exports[`App Reducer ERROR should work corectly 1`] = `
Object {
  "error": "Some thing wrong happened",
}
`;

exports[`App Reducer OPEN_TOAST should work corectly 1`] = `
Object {
  "toast": Object {
    "isDisplayToast": true,
    "success": true,
    "toastMsg": "successfully save user",
  },
}
`;

exports[`App Reducer default should work corectly 1`] = `Object {}`;
